src/core/src/package_managers/YumPackageManager.py

```diff
@@ -345,28 +371,331 @@ def get_package_size(self, output):
+        self.packagekit_config_pattern_match_text = ' = (false|true)'
+        self.packagekit_download_updates_identifier_text = 'GetPreparedUpdates'  # todo: dummy value, get real value
+        self.packagekit_apply_updates_identifier_text = 'WritePreparedUpdates'
+        self.packagekit_enable_on_reboot_identifier_text = "enable_on_reboot"
+
     def get_current_auto_os_patch_state(self):
         """ Gets the current auto OS update patch state on the machine """
-        # NOTE: Implementation pending
-        pass
+        self.composite_logger.log("Fetching the current automatic OS patch state on the machine...")
+
+        # This function can be called for a specific auto update service (when called from disable auto OS updates code flow), or for all auto OS update services (when called from ConfigurePatchingProcessor to fetch status)
+        # apply_updates_identifier_text will be initialized only when this function is called for a specific auto update service. If not initialized, we need to check status from all services
+        # todo: revisit this, we need to check apply_updates for each service and decide on enable or disable based on all statuses
+        if self.apply_updates_identifier_text == "":
+            for os_auto_update_service in self.auto_os_update_service_list:
+                if os_auto_update_service == self.yum_cron:
+                    self.composite_logger.log_debug("Fetching current automatic OS patch state in yum-cron service. This includes checks on current enable state and whether it is set to enable on reboot")
+                    self.__init_auto_update_for_yum_cron()
+                    self.__get_current_auto_os_updates_setting_on_machine()
+                    if self.enable_on_reboot_value or self.apply_updates_value.lower() == 'yes':
+                        break
+                elif os_auto_update_service == self.dnf_automatic:
+                    self.composite_logger.log_debug("Fetching current automatic OS patch state in dnf-automatic service. This includes checks on current state and whether it is set to enable on reboot")
+                    self.__init_auto_update_for_dnf_automatic()
+                    self.__get_current_auto_os_updates_setting_on_machine()
+                    if self.enable_on_reboot_value or self.apply_updates_value.lower() == 'yes':
+                        break
+                elif os_auto_update_service == self.packagekit:
+                    self.composite_logger.log_debug("Fetching current automatic OS patch state in packagekit service. This includes checks on current state and whether it is set to enable on reboot")
+                    self.__init_auto_update_for_packagekit()
+                    self.__get_current_auto_os_updates_setting_on_machine()
+                    if self.enable_on_reboot_value or self.apply_updates_value.lower() == 'true':
+                        break
```

Comment:
Still don't get this for loop. Every condition in here is guaranteed to run at least once.

Reply:
Resolved